### PR TITLE
support new honeywell simulator options

### PR DESCRIPTION
**Pull request by ss2165**
Simulator type (state-vector or stabilizer) specified on backend
initialisation, to perhaps support compilation and predicate checking in
future.
Whether to run noisy simulation specified at process_circuit time via
kwarg.

### CODE REVIEW

File: modules/pytket-honeywell/pytket/extensions/honeywell/backends/honeywell.py [hunk 1]
```diff
@@ -88,6 +88,10 @@
 }
 
 
+class GetResultFailed(Exception):
```
Comment (cqc-melf) [inline modules/pytket-honeywell/pytket/extensions/honeywell/backends/honeywell.py:91]: Is this the intended function?

File: modules/pytket-honeywell/pytket/extensions/honeywell/backends/honeywell.py [hunk 2]
```diff
@@ -61,7 +61,7 @@
 
 HONEYWELL_URL_PREFIX = "https://qapi.honeywell.com/v1/"
 
-HONEYWELL_DEVICE_APIVAL = "HQS-LT-1.0-APIVAL"
+HONEYWELL_DEVICE_APIVAL = "HQS-LT-S1-APIVAL"
```
Comment (cqc-melf) [inline modules/pytket-honeywell/pytket/extensions/honeywell/backends/honeywell.py:64]: This variable is unused. Is that intended ?

File: modules/pytket-honeywell/pytket/extensions/honeywell/backends/honeywell.py (hunk 1 above)
Comment (ss2165) [reply modules/pytket-honeywell/pytket/extensions/honeywell/backends/honeywell.py:91]: yes

File: modules/pytket-honeywell/pytket/extensions/honeywell/backends/honeywell.py (hunk 2 above)
Comment (ss2165) [reply modules/pytket-honeywell/pytket/extensions/honeywell/backends/honeywell.py:64]: no, will remove

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**